src/workerd/jsg/value.h

```diff
@@ -468,20 +469,25 @@ class StringWrapper {
       kj::Maybe<v8::Local<v8::Object>> parentObject) {
     // TODO(cleanup): Move to a HeaderStringWrapper in the api directory.
     v8::Local<v8::String> str = check(handle->ToString(context));
-    auto result =
-        ByteString(KJ_ASSERT_NONNULL(tryUnwrap(context, str, (kj::String*)nullptr, parentObject)));
-    if (str->ContainsOnlyOneByte()) {
-      auto buf = kj::heapArray<kj::byte>(str->Length() + 1);
-      str->WriteOneByte(context->GetIsolate(), buf.begin());
-      for (auto b: buf) {
-        if (b >= 128) {
-          result.warning = ByteString::Warning::CONTAINS_EXTENDED_ASCII;
-          break;
-        }
+    auto result = ByteString(KJ_ASSERT_NONNULL(
+        tryUnwrap(context, str, static_cast<kj::String*>(nullptr), parentObject)));
+
+    if (!simdutf::validate_ascii(result.begin(), result.size())) {
+      v8::String::ValueView str_view(context->GetIsolate(), str);
+
+      // If storage is one-byte or the string contains only one-byte
+      // characters, we know that it contains extended ASCII characters.
+      //
+      // The order of execution matters, since ContainsOnlyOneByte()
+      // will scan the whole string for two-byte storage.
+      if (str_view.is_one_byte() || str->ContainsOnlyOneByte()) {
```

Comment:
v8::String has IsOneByte() and ContainsOnlyOneByte() methods – can we elide the ValueView here?

Reply:
According to the documentation: v8::String IsOneByte() can return false negatives.

Reply:
Does the same not apply for ValueView too? There is no documentation for the is_one_byte() method, looking at the String::ValueView::ValueView definition the is_one_byte_ variable is set to `flat_content.IsOneByte();` so it should both be the same value unless the flattening eliminates false negatives here.

Reply:
I'm not sure.

Reply:
Another thing: ContainsOnlyOneByte will not always scan the whole thing – it performs the equivalent of the IsOneByte() check and returns early if it is in one-byte representation. That means we can get rid of the first condition entirely assuming that ValueView has the same false positives, ContainsOnlyOneByte will return without checking the string itself in every case where IsOneByte() applies.
Do we have reason to assume that ValueView avoids these false positives? I don't think so, and if it does I don't see how it would do so more cheaply than ContainsOnlyOneByte does without needing the same amount of computation.
TLDR: My understanding is the following: IsOneByte() returns if the string is known to be one-byte, if it is not known if it is or not it returns false (hence "false negatives"). ContainsOnlyOneByte() first checks if it is known to be such, if not it scans the string to find out. ValueView::is_one_byte() is functionally equivalent to IsOneByte(), it does not perform such scanning.

Reply:
Agreed. Removed the ValueView initialization. Thank you for the review @fhanau